CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+# Contributing
+
+- [Contributor Guidelines and Governance](#contributor-guidelines-and-governance)
+- [Pre-built images](#pre-built-images)
+- [Build Requirements](#build-requirements)
+- [Building Executables and Images from Scratch](#building-executables-and-images)
+- [Development](#development)
+- [Local Testing](#local-testing)
+
+## Contributor Guidelines and Governance
+
+Please see [CONTRIBUTING](https://github.com/spiffe/spiffe/blob/main/CONTRIBUTING.md) and [GOVERNANCE](https://github.com/spiffe/spiffe/blob/main/GOVERNANCE.md) from the SPIFFE project. 
+
+## Pre-built images
+
+You can use pre-built images for various versions and Tornjak components. For a list of supported public images of Tornjak along with usage instructions please see our [USAGE document](./USAGE.md).
+
+Otherwise, you can follow instructions below to build Tornjak images. 
+
+## Build Requirements
+
+In order to build, we require the following installations:
+- [Docker]() for the backend
+- [npm]() for the frontend
+
+## Building Executables and Images
+
+Building Tornjak manually can be done with the Makefile. Notable make targets follow:
+- `make bin/tornjak-backend`: makes the Go executable of the Tornjak backend
+- `make bin/tornjak-manager`: makes the Go executable of the Tornjak manager
+- `make ui-agent`: makes the ReactJS app for the Tornjak frontend
+- `make container-tornjak-be`: containerizes Go executable of the Tornjak backend
+- `make container-manager`:containerizes Go executable of the Tornjak manager
+- `make container-frontend`: containerizes React JS app for the Tornjak frontend
+
+For usage instructions of the containers, please see our [USAGE document](./USAGE.md) to get started.
+
+## Development
+
+We welcome all development attempst and contributions from the community. The easiest place to start is by reviewing our code architecture diagrams available in our [api documentation](./docs/tornjak-ui-api-documentation.md#11-overview).
+
+## Local testing
+
+We highly recommend starting with our [quickstart tutorial](./docs/tornjak-quickstart.md), using official images and preset configs before development. This tutorial creates a local instance of SPIRE on Minikube, adds Tornjak server, and runs a UI. 
+
+Additionally, one may test out several other features including the following:
+- [Running the Frontend Locally](#running-the-frontend-locally)
+- [Running the Tornjak Manager Backend and Frontend Locally](#running-the-tornjak-manager)
+- [User Management](#user-management)
+
+### Running the Frontend Locally
+
+An excellent feature for frontend development is the ability to make changes to the frontend code without needing to restart the application.  To start a local version of the Tornjak frontend, one must have it point at the running Tornjak APIs:
+
+```console
+cd tornjak-frontend
+REACT_APP_API_SERVER_URI=http://<tornjak_API>/  npm start
+```
+
+Assuming `npm` is installed, this will start a server on `http://localhost:3000`
```

Comment:
Please include a link where to the place and instructions where they can get `npm`.